addons/binding/org.openhab.binding.plugwise/src/main/java/org/openhab/binding/plugwise/handler/PlugwiseRelayDeviceHandler.java

```diff
@@ -0,0 +1,576 @@
+                            pendingPowerStateChange.onOff, pendingPowerStateChange.retries);
+                    handleOnOffCommand(pendingPowerStateChange.onOff);
+                } else {
+                    logger.warn("Failed to switch {} ({}) {} after {} retries", deviceType, macAddress,
+                            pendingPowerStateChange.onOff, pendingPowerStateChange.retries);
+                    pendingPowerStateChange = null;
+                }
+            }
+        }
+
+        if (pendingPowerStateChange == null) {
+            updateState(CHANNEL_STATE, powerState);
+            correctPowerState(powerState);
+        }
+
+        if (energy == null && isCalibrated()) {
+            updateEnergy();
+        }
+
+        updateProperties(message);
+    }
+
+    private void handleOnOffCommand(OnOffType command) {
+        sendCommandMessage(new PowerChangeRequestMessage(macAddress, command == OnOffType.ON));
+        sendFastUpdateMessage(new InformationRequestMessage(macAddress));
+
+        // Measurements take 2 seconds to become stable
+        scheduler.schedule(new Runnable() {
+            @Override
+            public void run() {
+                sendFastUpdateMessage(new PowerInformationRequestMessage(macAddress));
+            }
+        }, 2, TimeUnit.SECONDS);
+    }
+
+    private void handlePowerBufferResponse(PowerBufferResponseMessage message) {
+        if (!isCalibrated()) {
+            calibrate();
+            return;
+        }
+
+        Energy mostRecentEnergy = message.getMostRecentDatapoint();
+
+        if (mostRecentEnergy != null) {
+            // When the current time is '11:44:55.888' and the measurement interval 1 hour, then the end of the most
+            // recent energy measurement interval is at '11:00:00.000'
+            LocalDateTime oneIntervalAgo = LocalDateTime.now().minusMinutes(configuration.getMeasurementInterval());
+
+            boolean isLastInterval = mostRecentEnergy.getEnd().isAfter(oneIntervalAgo);
+            if (isLastInterval) {
+                energy = mostRecentEnergy;
+                energy.setInterval(configuration.getMeasurementInterval() * 60);
+                logger.trace("Updating {} ({}) energy with: {}", deviceType, macAddress, mostRecentEnergy);
+                updateState(CHANNEL_ENERGY, new DecimalType(correctSign(energy.tokWh(calibration))));
+                updateState(CHANNEL_ENERGY_STAMP, PlugwiseUtils.newDateTimeType(energy.getStart()));
+            }
+        }
+    }
+
+    private void handlePowerInformationResponse(PowerInformationResponseMessage message) {
+        if (!isCalibrated()) {
+            calibrate();
+            return;
+        }
+
+        Energy one = message.getOneSecond();
+        double watt = one.toWatt(calibration);
+        if (watt > 10000) {
+            logger.debug("{} ({}) is in a kind of error state, skipping power information response", deviceType,
```

Comment:
It just occasionally reports erroneous values eventhough everything else still works. So the binding just skips these when they do not make any sense.